README.md

```diff
@@ -7,59 +7,411 @@
+previous max entry is returned. If `entry.value` is not larger than the
+previous max entry's value, then the max entry remains the same and `entry` is
+returned back.
+
+#### Parameters
+
+- `entry` (Object) entry to add to the max value tracker.
+- `entry.value` (Integer String) value to compare to the current max value.
+- `entry.data` (Object) data to attach to the entry.
+
+#### Return
+
+- `return` (Object) previous max entry or `entry`.
+- `return.value` (Integer String) value of returned entry.
+- `return.data` (Object) data attached to returned entry.
+
+-------
+
+### `async MaxValueTracker.getMax ()`
+
+Returns the max value tracker's maximum entry.
+
+#### Return
+
+- `return` (Object) max entry of the max value tracker.
+- `return.value` (Integer String) value of returned entry.
+- `return.data` (Object) data attached to returned entry.
+
+## Payment Channel Backend API
+
+Calling `MakePluginVirtual` with an object containing some or all of the
+functions defined below will return a class. This new class will perform all the
+functionality of ILP Plugin Virtual, and additionally use the supplied callbacks
+to handle settlement.
+
+Aside from `connect` and `disconnect`, the functions below might be called
+during the flow of an RPC request. They should not take enough time to time out
+an HTTP request.
+
+-------
+
+### `async connect (ctx, opts)`
+
+Called when `plugin.connect()` is called.
+
+#### Parameters
+
+- `ctx` (PluginContext) current plugin context.
+- `opts` (Object) options passed into plugin constructor.
+
+-------
+
+### `async disconnect (ctx)`
+
+Called when `plugin.disconnect()` is called.
+
+#### Parameters
+
+- `ctx` (PluginContext) current plugin context.
+
+-------
+
+### `async handleIncomingPrepare (ctx, transfer)`
+
+Called when an incoming transfer is being processed, but has not yet been
+prepared. If this function throws an error, the transfer will not be prepared
+and the error will be passed back to the peer. 
+
+#### Parameters
+
+- `ctx` (PluginContext) current plugin context.
+- `transfer` (Transfer) incoming transfer.
+
+-------
+
+### `async createOutgoingClaim (ctx, balance)`
+
+Called when settlement is triggered. This may occur in the flow of a single payment,
+or it may occur only once per several payments. The return value of this function is
+passed to the peer, and into their `handleIncomingClaim()` function. The return value must
+be stringifiable to JSON.
+
+#### Parameters
+
+- `ctx` (PluginContext) current plugin context.
+- `balance` (Integer String) sum of all outgoing fulfilled transfers. This value is strictly increasing.
+
+-------
+
+### `async handleIncomingClaim (ctx, claim)`
+
+Called after peer's `createOutgoingClaim()` function is called.
+
+#### Parameters
+
+- `ctx` (PluginContext) current plugin context.
+- `claim` (Object) return value of peer's `createOutgoingClaim()` function.
+
+## Plugin Context API
```

Comment:
Nit: put the context API docs before the `ctx` is referenced in the function docs